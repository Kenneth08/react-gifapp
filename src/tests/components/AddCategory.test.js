import { AddCategory } from "../../components/AddCategory";
import { shallow } from "enzyme";

describe("Tests into AddCategory.js", () => {
  const setCategories = jest.fn();
  let wrapper = shallow(<AddCategory setCategories={setCategories} />);
  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test("should show <AddCategory /> successfully", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("should change input text", () => {
    const input = wrapper.find("input");
    const value = "Hola Mundo";

    input.simulate("change", { target: { value } });
  });
  test(`shouldn't post the info`, () => {
    wrapper.find("form").simulate("submit", { preventDefault() {} });

    expect(setCategories).not.toHaveBeenCalled();
  });
  test(`should call setCategories and clear the input`, () => {
    const value = "Hola Mundo";

    wrapper.find("input").simulate("change", { target: { value } });
    wrapper.find("form").simulate("submit", { preventDefault() {} });
    expect(setCategories).toHaveBeenCalled();
    expect(wrapper.find("input").prop("value")).toBe("");
  });
});
