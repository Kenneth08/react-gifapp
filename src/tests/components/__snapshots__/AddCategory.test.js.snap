// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests into AddCategory.js should show <AddCategory /> successfully 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Search gifs..."
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
